src/Features/Core/Portable/EmbeddedLanguages/Json/JsonHelpers.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.Common;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.Json
+{
+    using JsonToken = EmbeddedSyntaxToken<JsonKind>;
+    using JsonTrivia = EmbeddedSyntaxTrivia<JsonKind>;
+
+    internal static class JsonHelpers
+    {
+        public static JsonToken CreateToken(
+            JsonKind kind, ImmutableArray<JsonTrivia> leadingTrivia,
+            VirtualCharSequence virtualChars, ImmutableArray<JsonTrivia> trailingTrivia)
+            => CreateToken(kind, leadingTrivia, virtualChars, trailingTrivia, ImmutableArray<EmbeddedDiagnostic>.Empty);
+
+        public static JsonToken CreateToken(JsonKind kind,
+            ImmutableArray<JsonTrivia> leadingTrivia, VirtualCharSequence virtualChars,
+            ImmutableArray<JsonTrivia> trailingTrivia, ImmutableArray<EmbeddedDiagnostic> diagnostics)
+            => new(kind, leadingTrivia, virtualChars, trailingTrivia, diagnostics, value: null!);
+
+        public static JsonToken CreateMissingToken(JsonKind kind)
+            => CreateToken(kind, ImmutableArray<JsonTrivia>.Empty, VirtualCharSequence.Empty, ImmutableArray<JsonTrivia>.Empty);
```

Comment:
the virtual chars do expose their original span.  This is how we do things like classification, where we classify the correct original span of text given some final cahracter fed into json/regex. 

not sure i understand the rest of the question.

Reply:
@CyrusNajmabadi I think in the Roslyn model a missing token has a zero-width span, but it's start location is where the token "Is"; this helper has no way to take a span. My worry here is if some feature starts creating zero-width tags/classifications/whatever at the start of the file because the span here is going to be effectively always at the start.